#! /bin/sh
patch -p1 -l -f $* < $0
exit $?

From 6953b12e35a374dd85192def481575fc24190918 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ingo=20Kl=C3=B6cker?= <dev@ingo-kloecker.de>
Date: Wed, 19 Feb 2025 11:23:34 +0100
Subject: [PATCH] On Windows, embed our app icon as Breeze icon

Moreover, don't install the app icons on Windows because kiconthemes
only looks at embedded icons anyway.

GnuPG-bug-id: 7509
---
 src/CMakeLists.txt     | 15 ++++++++++++---
 src/icons/icons.qrc.in |  6 ++++++
 2 files changed, 18 insertions(+), 3 deletions(-)
 create mode 100644 src/icons/icons.qrc.in

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index b9d2b4f3a..17997a53d 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -580,9 +580,18 @@ kconfig_add_kcfg_files(_kleopatra_SRCS
 if (NOT KLEOPATRA_ICON_DIR)
     set(KLEOPATRA_ICON_DIR "${CMAKE_CURRENT_SOURCE_DIR}/icons")
 endif()
-file(GLOB ICONS_SRCS "${KLEOPATRA_ICON_DIR}/*.png" "${KLEOPATRA_ICON_DIR}/*.svg*")
-ecm_add_app_icon(_kleopatra_SRCS ICONS ${ICONS_SRCS})
-ecm_install_icons(ICONS ${ICONS_SRCS} DESTINATION ${KDE_INSTALL_ICONDIR})
+file(GLOB ICONS_PNGS "${KLEOPATRA_ICON_DIR}/*.png")
+file(GLOB ICONS_SVGS "${KLEOPATRA_ICON_DIR}/*.svg")
+ecm_add_app_icon(_kleopatra_SRCS ICONS ${ICONS_PNGS} ${ICONS_SVGS})
+if (NOT WIN32)
+    ecm_install_icons(ICONS ${ICONS_PNGS} ${ICONS_SVGS} DESTINATION ${KDE_INSTALL_ICONDIR})
+else()
+    if (ICONS_SVGS)
+        list(GET ICONS_SVGS 0 app_icon_svg)
+        configure_file(icons/icons.qrc.in icons.qrc @ONLY)
+        set(_kleopatra_SRCS ${_kleopatra_SRCS} ${CMAKE_CURRENT_BINARY_DIR}/icons.qrc)
+    endif()
+endif()

 add_executable(kleopatra_bin ${_kleopatra_SRCS} ${_kleopatra_uiserver_SRCS})

diff --git a/src/icons/icons.qrc.in b/src/icons/icons.qrc.in
new file mode 100644
index 000000000..1d3465784
--- /dev/null
+++ b/src/icons/icons.qrc.in
@@ -0,0 +1,6 @@
+<!DOCTYPE RCC>
+<RCC version="1.0">
+    <qresource prefix="/icons/breeze">
+        <file alias="apps/48/kleopatra.svg">@app_icon_svg@</file>
+    </qresource>
+</RCC>
--
2.48.1
